packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -935,7 +936,7 @@ export default function useAutocomplete(props) {
 
     if (autoSelect && highlightedIndexRef.current !== -1 && popupOpen) {
       selectNewValue(event, filteredOptions[highlightedIndexRef.current], 'blur');
-    } else if (autoSelect && freeSolo && inputValue !== '') {
+    } else if ((autoSelect || saveOnBlur) && freeSolo && inputValue !== '') {
```

Comment:
@sai6855 This makes me think why not use `autoSelect`?

Reply:
In reality, using `autoSelect` to save values on blur works, but description on `autoSelect` in docs says, `autoSelect` adds highlighted option to value. docs nowhere indicates `autoSelect` adds `inputValue` to `value`. so maybe should we update docs instead of adding new prop?
<img width="793" alt="Screenshot 2023-02-21 at 2 32 11 PM" src="https://user-images.githubusercontent.com/60743144/220297971-e9140ca6-dba7-4ecc-9255-6daeba29c3b4.png">

Reply:
If we go in updation of `autoSelect` prop description direction, it would be great if you can provide appropriate description, just to avoid further confusion

Reply:
From what I tested, adding a description sounds better than adding one more prop that likely does the same thing.

What do you think about this:

```ts
/**
* If `true`, the selected option becomes the value of the input
* when the Autocomplete loses focus unless the user chooses
* a different option or changes the character string in the input.
* 
* In addition of `freeSolo` mode, the selected option will the input value
* if the Autocomplete loses focus without highlighing an option.
* @default false
*/
autoSelect?: boolean;
```

Reply:
Sounds good!